stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/DatabaseChannelRepository.kt

```diff
@@ -258,8 +211,26 @@ internal class DatabaseChannelRepository(
                 channel.apply {
                     lastMessageAt = messageCreatedAt
                     messages = listOf(lastMessage)
-                }.also { insertChannel(it) }
+                }.also { upsertChannel(it) }
+            }
+        }
+    }
+
+    private suspend fun Channel.convertToEntity(): ChannelEntity {
+        val dbChannel = channelDao.select(this.cid)
+
+        return if (dbChannel?.lastMessageAt?.after(this.lastMessage?.createdAt ?: Date(Long.MIN_VALUE)) == true) {
```

Comment:
I'm curious if we also need to take into account `this.lastMessageAt` before hitting a fallback.
Plus I realized that probably should have used `Date(0)` as a min value, cause `Date(Long.MIN_VALUE)` may produce some future date.
```suggestion
        val thisLastMessageAt = this.lastMessage?.createdAt ?: this.lastMessageAt ?: Date(0)
        return if (dbChannel?.lastMessageAt?.after(thisLastMessageAt) == true) {
```